mlflow/pyfunc/__init__.py

```diff
@@ -847,7 +848,23 @@ def predict(*args):
         else:
             return result[result.columns[0]]
 
-    return pandas_udf(predict, result_type)
+    udf = pandas_udf(predict, result_type)
+
+    @functools.wraps(udf)
+    def udf_with_default_cols(*args):
+        if len(args) == 0:
+            model = SparkModelCache.get_or_load(archive_path)
+            input_schema = model.metadata.get_input_schema()
+
+            if input_schema and input_schema.has_input_names():
+                input_names = input_schema.input_names()
+                return udf(*input_names)
+            else:
+                return udf(*args)
```

Comment:
I guess we should just remove `*args` here since it's an empty array.

Reply:
I see. Can we catch the missing argument error and display the message then? I am worried that there is no way to tell if you can call the model udf with no arguments and it can be confusing. If we at least can inform the user that this particular model has no schema or no column names that would be helpful.

Reply:
Sounds good. The missing argument error is actually raised by the `predict` function defined a few lines above. Instead of catching the error and displaying a different message, does it make sense if we just clarify the error message if the model has no schema and no col name arguments (line 840 in the diff) ? Let me know if you actually meant something different.